gnd/src/main/java/com/google/android/gnd/system/CameraManager.java

```diff
@@ -50,19 +51,17 @@ public Completable launchImageCapture() {
         .andThen(sendCaptureImageIntent());
   }
 
-  public Observable<Optional<Bitmap>> captureImageResult() {
+  public Observable<Bitmap> captureImageResult() {
     return activityStreams
         .getNextActivityResult(CAPTURE_PHOTO_REQUEST_CODE)
         .filter(ActivityResult::isOk)
         .map(
-            activityResult -> {
-              Intent intent = activityResult.getData();
-              if (intent != null && intent.getExtras() != null) {
-                Bitmap photo = (Bitmap) intent.getExtras().get("data");
-                return Optional.ofNullable(photo);
-              }
-              return Optional.empty();
-            });
+            activityResult ->
+                Optional.ofNullable(activityResult.getData())
+                    .map(Intent::getExtras)
+                    .map(extras -> (Bitmap) extras.get("data")))
+        .filter(Optional::isPresent)
```

Comment:
If we don't need to return `Optional`, then this can be further simplified to remove the need to wrap and unwrap in `Optional`, e.g.:





Sorry!